File: android/versioned-abis/expoview-abi37_0_0/src/main/java/abi37_0_0/expo/modules/sensors/SensorsPackage.java
package abi37_0_0.expo.modules.sensors;

import android.content.Context;

import java.util.Arrays;
import java.util.List;

import abi37_0_0.org.unimodules.core.BasePackage;
import abi37_0_0.org.unimodules.core.ExportedModule;
import abi37_0_0.org.unimodules.core.interfaces.InternalModule;
import abi37_0_0.expo.modules.sensors.modules.AccelerometerModule;
import abi37_0_0.expo.modules.sensors.modules.BarometerModule;
import abi37_0_0.expo.modules.sensors.modules.DeviceMotionModule;
import abi37_0_0.expo.modules.sensors.modules.GyroscopeModule;
import abi37_0_0.expo.modules.sensors.modules.MagnetometerModule;
import abi37_0_0.expo.modules.sensors.modules.MagnetometerUncalibratedModule;
import abi37_0_0.expo.modules.sensors.modules.PedometerModule;
import abi37_0_0.expo.modules.sensors.services.AccelerometerService;
import abi37_0_0.expo.modules.sensors.services.BarometerService;
import abi37_0_0.expo.modules.sensors.services.GravitySensorService;
import abi37_0_0.expo.modules.sensors.services.GyroscopeService;
import abi37_0_0.expo.modules.sensors.services.LinearAccelerationSensorService;
import abi37_0_0.expo.modules.sensors.services.MagnetometerService;
import abi37_0_0.expo.modules.sensors.services.MagnetometerUncalibratedService;
import abi37_0_0.expo.modules.sensors.services.PedometerService;
import abi37_0_0.expo.modules.sensors.services.RotationVectorSensorService;

public class SensorsPackage extends BasePackage {

  @Override
  public List<InternalModule> createInternalModules(Context context) {
    return Arrays.<InternalModule>asList(
            new AccelerometerService(context),
            new BarometerService(context),
            new GravitySensorService(context),
            new GyroscopeService(context),
            new LinearAccelerationSensorService(context),
            new MagnetometerService(context),
            new MagnetometerUncalibratedService(context),
            new RotationVectorSensorService(context),
            new PedometerService(context)
    );
  }

  @Override
  public List<ExportedModule> createExportedModules(Context context) {
    return Arrays.<ExportedModule>asList(
            new AccelerometerModule(context),
            new BarometerModule(context),
            new GyroscopeModule(context),
            new DeviceMotionModule(context),
            new MagnetometerModule(context),
            new MagnetometerUncalibratedModule(context),
            new PedometerModule(context)
    );
  }
}

File: android/versioned-abis/expoview-abi45_0_0/src/main/java/abi45_0_0/expo/modules/sensors/SensorsPackage.kt
package abi45_0_0.expo.modules.sensors

import android.content.Context
import abi45_0_0.expo.modules.sensors.modules.AccelerometerModule
import abi45_0_0.expo.modules.sensors.modules.BarometerModule
import abi45_0_0.expo.modules.sensors.modules.DeviceMotionModule
import abi45_0_0.expo.modules.sensors.modules.GyroscopeModule
import abi45_0_0.expo.modules.sensors.modules.MagnetometerModule
import abi45_0_0.expo.modules.sensors.modules.MagnetometerUncalibratedModule
import abi45_0_0.expo.modules.sensors.modules.PedometerModule
import abi45_0_0.expo.modules.sensors.services.AccelerometerService
import abi45_0_0.expo.modules.sensors.services.BarometerService
import abi45_0_0.expo.modules.sensors.services.GravitySensorService
import abi45_0_0.expo.modules.sensors.services.GyroscopeService
import abi45_0_0.expo.modules.sensors.services.LinearAccelerationSensorService
import abi45_0_0.expo.modules.sensors.services.MagnetometerService
import abi45_0_0.expo.modules.sensors.services.MagnetometerUncalibratedService
import abi45_0_0.expo.modules.sensors.services.PedometerService
import abi45_0_0.expo.modules.sensors.services.RotationVectorSensorService
import abi45_0_0.expo.modules.core.BasePackage
import abi45_0_0.expo.modules.core.ExportedModule
import abi45_0_0.expo.modules.core.interfaces.InternalModule

class SensorsPackage : BasePackage() {
  override fun createInternalModules(context: Context): List<InternalModule> {
    return listOf<InternalModule>(
      AccelerometerService(context),
      BarometerService(context),
      GravitySensorService(context),
      GyroscopeService(context),
      LinearAccelerationSensorService(context),
      MagnetometerService(context),
      MagnetometerUncalibratedService(context),
      RotationVectorSensorService(context),
      PedometerService(context)
    )
  }

  override fun createExportedModules(context: Context): List<ExportedModule> {
    return listOf<ExportedModule>(
      AccelerometerModule(context),
      BarometerModule(context),
      GyroscopeModule(context),
      DeviceMotionModule(context),
      MagnetometerModule(context),
      MagnetometerUncalibratedModule(context),
      PedometerModule(context)
    )
  }
}
